src/sciline/pipeline.py

```diff
@@ -50,6 +53,66 @@ def _is_multiple_keys(keys: type | Iterable[type] | UnionType) -> bool:
     )
 
 
+def _find_paths_to_targets(
+    graph: Any, missing: Any, targets: tuple[Any, ...]
+) -> list[list[Any]]:
+    """Find paths from missing node to each target.
+
+    Parameters
+    ----------
+    graph:
+        The graph to traverse
+    missing:
+        The missing node
+    targets:
+        The target nodes
+
+    Returns
+    -------
+    :
+        List of paths, where each path is a list of nodes from missing to target
+    """
+
+    paths = []
+    for target in targets:
+        try:
+            # Find all simple paths from the missing node to the target
+            target_paths = list(all_simple_paths(graph, missing, target))
+            # Sort by length to show the shortest path first
+            paths.extend(sorted(target_paths, key=len))
+        except (nx.NetworkXNoPath, nx.NodeNotFound):  # noqa: PERF203
+            # No path found or nodes not in graph
+            continue
+
+    return paths
+
+
+def _format_paths_msg(nx_graph: Any, paths: list[list[Any]]) -> str:
+    """Format paths into a readable message.
+
+    Parameters
+    ----------
+    paths:
+        List of paths, where each path is a list of nodes
+
+    Returns
+    -------
+    :
+        The formatted message
+    """
+    msg = f"Missing input node '{key_name(paths[0][0])}'. "
+    msg += "Affects requested targets (via providers given in parentheses):"
+
+    for i, path in enumerate(paths):
+        msg += f"\n{i + 1}. {key_name(path[0])} → "
+        msg += " → ".join(
```

Comment:
I don't know how widely utf-8 is supported in terminals. Especially in embedded terminals in editors and Windows terminals. Can you replace the arrows with `->` or just `>`?

Reply:
Hmm, we have not avoided utf-8 so far, e.g., for formatting units with "micro"?

Reply:
Ok. It won't be fundamentally broken if this can't be rendered. So I am ok with leaving it.